### PR TITLE
New/hm/posts appear with newest post first

### PR DESCRIPTION
**Pull request by maddc0de**
I closed the previous pull request that me and Hande opened, and I made the following changes on the code:

- I just moved the sorting method to the backend - I included a comment on `api/controllers/posts.js` why - please let me know what you think..
This way, the response body will return a sorted posts data when the frontend makes a GET /posts request

- I test drive the changes in the response body on `api/spec/controllers/posts.spec.js` from line 90 - 108

### CODE REVIEW

**Comment by maddc0de** [comment]
- one more minor change: I renamed the field from `lastUpdated` to `dateCreated` in the `api/models/post.js` for clarity

File: api/controllers/posts.js [hunk 1]
```diff
@@ -7,7 +7,12 @@ const PostsController = {
       if (err) {
         throw err;
       }
-      const token = await TokenGenerator.jsonwebtoken(req.user_id)
+      const token = await TokenGenerator.jsonwebtoken(req.user_id);
+
+      // originally in frontend, but I thought frontend should have minimal code and main purpose is for displaying
```
Comment (toppy007) [inline api/controllers/posts.js]: Yeah looks good, nice idea. before merging you can remove your comments I think, but up to you

File: api/models/post.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 const mongoose = require("mongoose");
 
 const PostSchema = new mongoose.Schema({
-  message: String
+  message: { type: String },
+  dateCreated: { type: Date, default: Date.now },
```
Comment (toppy007) [inline api/models/post.js:5]: guess you've checked this but does this added parameter to the database affect any other test?

File: api/spec/controllers/posts.spec.js [hunk 3]
```diff
@@ -49,112 +52,124 @@ describe("/posts", () => {
...
-      let post1 = new Post({message: "howdy!"});
-      let post2 = new Post({message: "hola!"});
+    test("returns posts in order from newest to oldest", async () => {
+      let post1 = new Post({
+        message: "howdy!",
+        dateCreated: "2023-05-10T20:51:59.427Z",
+      });
+      let post2 = new Post({
+        message: "hola!",
+        dateCreated: "2023-05-10T22:51:59.427Z",
```
Comment (toppy007) [inline api/spec/controllers/posts.spec.js:101]: nice test double. but a few ideas!

Is it not worth give both posts a different time stamp? so you can use (expects) to pick the first one to check against?

Might be worth adding in the project documentation the format type for the dateCreated but its not that important.

File: api/spec/controllers/posts.spec.js [hunk 4]
```diff
@@ -49,112 +52,124 @@ describe("/posts", () => {
...
         .get("/posts")
         .set("Authorization", `Bearer ${token}`)
-        .send({token: token});
-      let messages = response.body.posts.map((post) => ( post.message ));
-      expect(messages).toEqual(["howdy!", "hola!"]);
-    })
+        .send({ token: token });
+      console.log(response.body.posts);
+      let messages = response.body.posts.map((post) => post.message);
+      expect(messages).toEqual(["hola!", "howdy!"]);
```
Comment (toppy007) [inline api/spec/controllers/posts.spec.js:111]: I think I understand the test but let me know if I'm wrong

guess you want to check that the first message is the one matching the correct time stamp!

not sure how to do that tho!!

File: api/spec/controllers/posts.spec.js [hunk 5]
```diff
@@ -49,112 +52,124 @@ describe("/posts", () => {
...
-    })
-  })
+    });
+  });
 });
+
+// test "GET, index returns posts in order"
+// test("does not return a new token", async () => {
+//   let post1 = new Post({ message: "howdy!" });
+//   let post2 = new Post({ message: "hola!" });
```
Comment (toppy007) [inline api/spec/controllers/posts.spec.js]: Guess this can be removed now?

File: frontend/src/components/feed/Feed.js [hunk 6]
```diff
@@ -1,49 +1,47 @@
...
   const [token, setToken] = useState(window.localStorage.getItem("token"));
 
   useEffect(() => {
-    if(token) {
+    if (token) {
       fetch("/posts", {
         headers: {
-          'Authorization': `Bearer ${token}`
-        }
+          Authorization: `Bearer ${token}`,
```
Comment (toppy007) [inline frontend/src/components/feed/Feed.js:13]: Not sure what is the syntax require at this part will let you deal with this.

**Review by toppy007 — COMMENTED** [review]
Very nice work.

recon you could have a look at the test you've written but maybe I've misread it.

It looks like you might have some merging issue but maybe that something we could work on as a team when your ready.

File: frontend/src/components/feed/Feed.js (hunk 6 above)
Comment (maddc0de) [reply frontend/src/components/feed/Feed.js:13]: its the prettier extension styling.. it added it there

File: api/spec/controllers/posts.spec.js (hunk 5 above)
Comment (maddc0de) [reply api/spec/controllers/posts.spec.js]: oops sorry forgot about this!

File: api/spec/controllers/posts.spec.js (hunk 4 above)
Comment (maddc0de) [reply api/spec/controllers/posts.spec.js:111]: im checking the order of the post messages in the array based on the time it was created

File: api/models/post.js (hunk 2 above)
Comment (maddc0de) [reply api/models/post.js:5]: it didn't affect the other tests I dont think Ive run the tests on both api and frontend

File: api/spec/controllers/posts.spec.js (hunk 3 above)
Comment (maddc0de) [reply api/spec/controllers/posts.spec.js:101]: they do have different time stamp, the first one was posted at 8pm and the second one was posted at 10pm

Comment (toppy007) [reply api/spec/controllers/posts.spec.js:101]: yeah nice sorry for missing that was pre coffee review!!!

**Review by toppy007 — APPROVED** [review]
yeah nice work. sorry about missing the timestamp difference.

let us know if you have any merge conflicts you don't understand. 

but approved by me

**Pull request merged**